runtime/ast/access.go

```diff
@@ -20,38 +20,167 @@ package ast
 
 import (
 	"encoding/json"
+	"strings"
 
 	"github.com/onflow/cadence/runtime/errors"
 )
 
-//go:generate go run golang.org/x/tools/cmd/stringer -type=Access
+//go:generate go run golang.org/x/tools/cmd/stringer -type=PrimitiveAccess
 
-type Access uint
+type Access interface {
+	isAccess()
+	Keyword() string
+	Description() string
+	String() string
+	MarshalJSON() ([]byte, error)
+	IsLessPermissiveThan(Access) bool
+}
+
+type EntitlementSet interface {
+	Entitlements() []*NominalType
+	Separator() string
```

Comment:
Noticed while checking the follow-up PRs: maybe return an enum (iota) for the separator, rather than a string.
Could avoid any inconsistencies, e.g: the `DisjunctiveEntitlementSet` separator has a preceding whitespace, but the `ConjunctiveEntitlementSet` does not.